Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs

```diff
@@ -0,0 +1,77 @@
+﻿using EFCoreAssignment.Data.Entities;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Storage.ValueConversion.Internal;
+
+namespace EFCoreAssignment.Data.Services
+{
+    public class ProductService : IProductService
+    {
+        private readonly AppDbContext _context;
+
+        public ProductService(AppDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<List<ProductDto>> GetProducts()
+        {
+            // TODO get products
+            var product = await _context.Products.ToListAsync();
+            return product.Select(p => new ProductDto(p.Id, p.Name, p.ShopId)).ToList();
+        }
+
+        public async Task<ProductDto> GetProduct(int id)
+        {
+            // TODO get product
+            var product = await _context.Products.FindAsync(id);
```

Comment:
You can use .Select method to directly cast the Product to ProductDto

Reply:
hello here is the updated code
![image](https://github.com/miguelrefuerzo/EFCoreMasters/assets/106952123/854745e6-24fa-4b39-85e0-bbd42ba663e2)